### PR TITLE
Update DG UI component and implementation of DoneCommand

### LINKED ISSUE #112: Update DG UI component and implementation of done command

### PR DESCRIPTION
**Pull request by YangYue128-helen**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -62,9 +62,10 @@ The sections below give more details of each component.
 ![Structure of the UI Component](images/UiClassDiagram.png)
 
 **API** :
-[`Ui.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/Ui.java)
+[`Ui.java`](https://github.com/se-edu/AY2021S1-CS2103T-T12-4/tp/tree/master/src/main/java/seedu/address/ui/Ui.java)
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `StatisticsDisplay`, `AppointmentListPanel`, 
+`PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
```
Comment (Avalionnet) [inline docs/DeveloperGuide.md]: Should `PersonListPanel` be `PatientListPanel` instead?

File: docs/diagrams/DoneSequenceDiagram.puml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+participant ":DoneCommandParser" as DoneCommandParser LOGIC_COLOR
+participant "loader:DateTimeLoader" as DateTimeLoader LOGIC_COLOR
+participant "d:DoneCommand" as DoneCommand LOGIC_COLOR
+end box
+
+box Model MODEL_COLOR_T1
+participant ":Model" as Model MODEL_COLOR
+participant "toMark:Appointment" as Appointment MODEL_COLOR
+participant "doneAppointment:Appointment" as DoneAppointment MODEL_COLOR
+participant ":VersionedPatientBook" as VersionedPatientBook MODEL_COLOR
```
Comment (JinHao-L) [inline docs/diagrams/DoneSequenceDiagram.puml]: I don't think there is a `VersionedPatientBook` here

File: docs/DeveloperGuide.md [hunk 3]
```diff
@@ -133,85 +134,50 @@ Classes used by multiple components are in the `seedu.patientbook.commons` packa
...
+#### 1.1 Implementation
+Command: `done d/DATE t/TIME`
 
-#### Proposed Implementation
+Example Usage: 
+* `done d/Monday t/9am`
+* `done d/12-12-2020 t/12pm`
 
-The proposed undo/redo mechanism is facilitated by `VersionedPatientBook`. It extends `PatientBook` with an undo/redo history, stored internally as an `patientBookStateList` and `currentStatePointer`. Additionally, it implements the following operations:
+The Mark Appointments as Done Feature is facilitated by the `DoneCommand`, which extends the abstract class `Command`, and
```
Comment (Avalionnet) [inline docs/DeveloperGuide.md]: Hmm perhaps the phrase "mark appointments" and "feature" should be in lower case?

File: docs/DeveloperGuide.md [hunk 4]
```diff
@@ -133,85 +134,50 @@ Classes used by multiple components are in the `seedu.patientbook.commons` packa
...
+the `DoneCommandParser`, which implements the `Parser` interface. Both of these classes are part of the `Logic` component.
+Additionally, the command takes in a `DateTimeLoader`, which contains the date and time information of the appointment to mark.
 
-* `VersionedPatientBook#commit()` — Saves the current patient book state in its history.
-* `VersionedPatientBook#undo()` — Restores the previous patient book state from its history.
-* `VersionedPatientBook#redo()` — Restores a previously undone patient book state from its history.
+The following is an example usage scenario and how the mark as done mechanism behaves at each step:
```
Comment (Avalionnet) [inline docs/DeveloperGuide.md]: Perhaps you may consider the following phrase instead "Please refer to the following for an example usage scenario on how the mark as done mechanism works in each step:". 

Let me know what you think! 😄

File: docs/diagrams/DoneSequenceDiagram.puml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+Appointment --> DoneCommand : toMark
+deactivate Appointment
+
+create DoneAppointment
+DoneCommand -> DoneAppointment : toMark.markAsDone()
+activate DoneAppointment
+
+DoneAppointment --> DoneCommand : doneAppointment
+deactivate DoneAppointment
+
```
Comment (JinHao-L) [inline docs/diagrams/DoneSequenceDiagram.puml]: The appointment should not be created when called with `lastShownAppointmentList.filter`.
`toMark.markAsDone` should call the `markAsDone` method in `toMark:Appointment` which in turns call the constructor of `doneAppointment`

File: docs/DeveloperGuide.md [hunk 6]
```diff
@@ -62,9 +62,10 @@ The sections below give more details of each component.
...
 
 **API** :
-[`Ui.java`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/Ui.java)
+[`Ui.java`](https://github.com/se-edu/AY2021S1-CS2103T-T12-4/tp/tree/master/src/main/java/seedu/address/ui/Ui.java)
 
-The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
+The UI consists of a `MainWindow` that is made up of parts e.g.`CommandBox`, `ResultDisplay`, `StatisticsDisplay`, `AppointmentListPanel`, 
+`PersonListPanel`, `StatusBarFooter` etc. All these, including the `MainWindow`, inherit from the abstract `UiPart` class.
 
 The `UI` component uses JavaFx UI framework. The layout of these UI parts are defined in matching `.fxml` files that are in the `src/main/resources/view` folder. For example, the layout of the [`MainWindow`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/java/seedu/address/ui/MainWindow.java) is specified in [`MainWindow.fxml`](https://github.com/se-edu/addressbook-level3/tree/master/src/main/resources/view/MainWindow.fxml)
```
Comment (JinHao-L) [inline docs/DeveloperGuide.md]: The link here should also be updated?

File: docs/DeveloperGuide.md [hunk 7]
```diff
@@ -133,85 +134,50 @@ Classes used by multiple components are in the `seedu.patientbook.commons` packa
...
-
-![UndoSequenceDiagram](images/UndoSequenceDiagram.png)
-
-<div markdown="span" class="alert alert-info">:information_source: **Note:** The lifeline for `UndoCommand` should end at the destroy marker (X) but due to a limitation of PlantUML, the lifeline reaches the end of diagram.
-
-</div>
+10. `DoneCommand` replaces the `toMark` with the `doneAppointment` by calling the `Model#setAppointment(target, editedAppointment)`.
 
-The `redo` command does the opposite — it calls `Model#redoPatientBook()`, which shifts the `currentStatePointer` once to the right, pointing to the previously undone state, and restores the patient book to that state.
+11. As a last step, `DoneCommand` creates a `CommandResult` with `SuccessMessage` and returns it to `LogicManager`.
```
Comment (Avalionnet) [inline docs/DeveloperGuide.md]: Maybe it'll improve the flow by replacing the phrase "As a last step" with "Lastly,".

Also, it seems to me that you missed out a "a" in  "`CommandResult` with a `SuccessMessage` ".

**Review by JinHao-L — COMMENTED** [review]
Other than the comments mentioned, LGTM 👍

File: docs/diagrams/DoneSequenceDiagram.puml [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+deactivate Model
+
+DoneCommand --> LogicManager : result
+deactivate DoneCommand
+DoneCommand -[hidden]-> LogicManager : result
+destroy DoneCommand
+
+[<--LogicManager
+deactivate LogicManager
+@enduml
```
Comment (Avalionnet) [inline docs/diagrams/DoneSequenceDiagram.puml:88]: Great job on the diagrams!! 😄

File: docs/diagrams/UiClassDiagram.puml [hunk 9]
```diff
@@ -32,28 +35,36 @@ UiManager .left.|> Ui
...
 
 MainWindow -left-|> UiPart
 
 ResultDisplay --|> UiPart
+StatisticsDisplay --|> UiPart
+AppointmentListPanel --|> UiPart
+AppointmentCard --|> UiPart
 CommandBox --|> UiPart
 PersonListPanel --|> UiPart
 PersonCard --|> UiPart
```
Comment (Avalionnet) [inline docs/diagrams/UiClassDiagram.puml]: Hmm, perhaps all instances of "Person" should be changed to "Patient" for the diagrams.

**Review by Avalionnet — COMMENTED** [review]
LGTM other than the few minor changes 👍🏻

**Review by JinHao-L — APPROVED** [review]
Need some further improvement on the sequence diagram, but this can be addressed in another PR.
LGTM! 👍

**Pull request merged**